allennlp/common/params.py

```diff
@@ -203,14 +188,18 @@ def _check_is_dict(self, new_history, value):
         return value
 
     @staticmethod
-    def from_file(params_file: str) -> 'Params':
+    def from_file(params_file: str, overrides: List[str] = []) -> 'Params':
         """
         Load a `Params` object from a configuration file.
         """
         # redirect to cache, if necessary
         params_file = cached_path(params_file)
 
-        param_dict = pyhocon.ConfigFactory.parse_file(params_file)
+        override_hocon = "{" + ",".join(overrides) + "}"
```

Comment:
yeah, I don't like this at all, let's try to think of a safer/cleaner way to do this

Reply:
What don't you like about it?  I can figure out how to use argparse and `from_dict`.

Reply:
something more like

```
override_hocon = pyhocon.ConfigTree()
for override in overrides:
  key, value = ... # whatever you need to do here
  override_hocon.put(key, value)
```

Reply:
I feel much better about something like this than about string.join -> parseString

Reply:
I can use `from_dict`, which will do about the same thing.  I will need to add my own parsing of the CLI arguments from argparse to separate them into key and value.  I'll use the following syntax.

```
python allennlp/run.py train -s output experiment_config/bidaf.json --overrides 'train_data_pat="foo"' 'validation_data_path="bar"'
```

As a disadvantage, this syntax is less flexible and requires more documentation.  Previously, the syntax was just "add HOCON overrides".

Reply:
When you're testing this, make sure you can override things that are nested a few levels in the parameters.

Reply:
I don't care which format you use, my point was more just that glomming a bunch of stuff into a string and then parsing it smells bad to me, compared to parsing the things individually and doing something with them. for example, if one of them is malformed, you get a much better error message if you're trying to understand each one.

Reply:
@matt-gardner I'll be sure to.  I have not written any tests yet.

@joelgrus thanks--I agree we'll get much better error messages with another approach.  Thanks--I just wanted to understand your precise concern.

Reply:
@matt-gardner formal tests to come, but `dataset_reader.token_indexers.token_characters.character_tokenizer.end_tokens: [999]` overrode fine.

Reply:
Great, I was just concerned that a naive dictionary assignment wouldn't work for this, and wanted to be sure it was on your radar.  Glad you figured it out.